airbyte-integrations/connectors/source-pendo-python/source_pendo_python/streams.py

```diff
@@ -189,6 +190,82 @@ def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
         return "report"
 
 
+class ReportResult(PendoPythonStream):
+    json_schema = None  # Field to store dynamically built Airbyte Stream Schema
+    primary_key = "reportId"
+
+    def __init__(self, report: Mapping[str, Any], **kwargs):
+        super().__init__(**kwargs)
+        self.report = report
+        self.report_name = f"report_result_{report['id']}"
+
+    @property
+    def name(self):
+        return self.report_name
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"report/{self.report['id']}/results.json"
+
+    def parse_response(
+        self,
+        response: requests.Response,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        for record in response.json():
+            yield self.transform(record=record)
+
+    def transform(self, record: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
+        record["reportId"] = self.report['id']
+        return record
+
+    # Method to infer schema types from JSON response
+    def infer_type(self, value: Any):
+        if isinstance(value, str):
+            return {"type": ["null", "string"]}
+        if isinstance(value, bool):
+            return {"type": ["null", "boolean"]}
+        if isinstance(value, float):
+            return {"type": ["null", "number"]}
+        if isinstance(value, int):
+            return {"type": ["null", "integer"]}
+        if isinstance(value, list):
+            return {"type": ["null", "array"]}
+        if isinstance(value, dict):
+            return {"type": ["null", "object"]}
+        return {"type": ["null", "string"]}
+
+    def get_json_schema(self) -> Mapping[str, Any]:
+        if self.json_schema is None:
+            schema = {
+                "type": "object",
+                "$schema": "http://json-schema.org/schema#",
+                "properties": {
+                    "reportId": {
+                        "type": "string"
+                    }
+                }
+            }
+
+            url = f"{PendoPythonStream.url_base}{self.path()}"
+            auth_headers = self.authenticator.get_auth_header()
+            try:
+                session = requests.get(url, headers=auth_headers)
+                body = session.json()
+                if body is not None and len(body) != 0:
+                    for result in body:
+                        for field in result:
+                            if result[field] is not None:
+                                schema["properties"][field] = self.infer_type(result[field])
+                self.json_schema = schema
+            except requests.exceptions.RequestException:
+                print("Error fetching sample Pendo Report Results")
```

Comment:
nit:  I'd suggest logging the specific exception / error message to make debugging / diagnostics easier.

will the pendo reports with errors (for example too old) trigger the exception block?  we want to make sure we log those so we understand what we at least attempted to fetch